### DISCUSSION TITLE
在这里讨论问题会不会比Issues更方便?

### DISCUSSION THREAD

**Discussion by teckick** [LearningOS/rust-based-os-comp2022#47] [Polls]

**Comment by jackyliu16** [comment]
我感觉要不先试试？

**Comment by teckick** [comment]
我自己的一点理解:

Issue 的终态应该都是 Closed, 但 Discussion 不是.

Github Issues 的筛选器可以同时筛选 Issues 和 Pull Request, 但是不能筛选 Discussions, 也可以理解为 Issue 和 PR 关系更加密切. 另外, 为 Issues 提供的默认 Label 中, 看起来只有 Question 与交流讨论有关, 其他的还是更侧重代码仓库.

相比 Issue 的基于 Label 的扁平化分类方式, Discussion 提供了 Category 和 Label 二级分类, 并且一些 Category 拥有特殊功能, 如投票.

**Comment by ChenMiaoi** [comment]
# Rustlings中所遇到的疑惑

- **可能会透露答案**。在rustlings中的一个问题，在iterator中，有一个函数
![-78bd5926acdfaa22.png](https://user-images.githubusercontent.com/96290967/228409639-62d70390-4cf6-4b39-90c2-e4bd47e97fa0.png)

![-1b89d5d6ccaa136a.png](https://user-images.githubusercontent.com/96290967/228409669-711eb899-3228-4af1-bab3-f1ac7aff027e.png)



如上图一样，我分开写就能够正常通过，但合在一起就会报关于生命周期的错误（由于原因，目前没能截图）。不知道是什么导致的😭

**Reply by MoonOld** [reply]
虽然但是，这么写的问题应该是泛型推断的问题吧，collect生成的是泛型，而extend特征也是泛型，rustc在静态编译的时候需要确定或者推导出确切的类型才可以找到正确的方法进行调用。这里extend直接拿走了Vec的所有权，并不涉及生命周期的问题。如果使用RA的话应该可以直接看到“type annotations needed”报错，或者你直接跑cargo check也可以。
对于第一段代码，改成
``` rust
s.extend(words.into_iter()
                .copied()
                .map(capitalize_first)
                .collect::<Vec<String>>());
```
即可解决。

**Reply by ChenMiaoi** [reply]
确实是，谢谢谢谢
